apps/OboeTester/app/src/main/cpp/NativeAudioContext.cpp

```diff
@@ -632,6 +632,46 @@ void ActivityRoundTripLatency::finishOpen(bool isInput, AudioStream *oboeStream)
     }
 }
 
+// The timestamp latency is the difference between the input
+// and output times for a specific frame.
+// Start with the position and time from an input timestamp.
+// Map the input position to the corresponding position in output
+// and calculate its time.
+// Use the difference between framesWritten and framesRead to
+// convert input positions to output positions.
+jdouble ActivityRoundTripLatency::measureTimestampLatency() {
+    if (!mFullDuplexLatency->isWriteReadDeltaValid()) return -1.0;
+
+    int64_t writeReadDelta = mFullDuplexLatency->getWriteReadDelta();
+    auto inputTimestampResult = mFullDuplexLatency->getInputStream()->getTimestamp(CLOCK_MONOTONIC);
+    if (!inputTimestampResult) return -1.0;
+    auto outputTimestampResult = mFullDuplexLatency->getOutputStream()->getTimestamp(CLOCK_MONOTONIC);
+    if (!outputTimestampResult) return -1.0;
+
+    int64_t inputPosition = inputTimestampResult.value().position;
+    int64_t inputTimeNanos = inputTimestampResult.value().timestamp;
+    int64_t ouputPosition = outputTimestampResult.value().position;
+    int64_t outputTimeNanos = outputTimestampResult.value().timestamp;
+
+    // Map input frame position to the corresponding output frame.
+    int64_t mappedPosition = inputPosition + writeReadDelta;
+    // Calculate when that frame will play.
+    int32_t sampleRate = mFullDuplexLatency->getOutputStream()->getSampleRate();
+    int64_t mappedTimeNanos = outputTimeNanos + ((mappedPosition - ouputPosition) * 1e9) / sampleRate;
+
+//    LOGD("ActivityRoundTripLatency::%s() writeReadDelta = %ld ", __func__, writeReadDelta);
+//    LOGD("ActivityRoundTripLatency::%s() inputPosition = %ld ", __func__, inputPosition);
+//    LOGD("ActivityRoundTripLatency::%s() inputTimeNanos = %ld ", __func__, inputTimeNanos);
```

Comment:
Should we remove comments?

Reply:
I am not convinced the algorithm is correct. So I occasionally stare at these numbers. I made them conditional.